pkg/notifications/shoutrrr.go

```diff
@@ -0,0 +1,88 @@
+package notifications
+
+import (
+	"fmt"
+	"github.com/containrrr/shoutrrr"
+	t "github.com/containrrr/watchtower/pkg/types"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const (
+	shoutrrrType = "shoutrrr"
+)
+
+// Implements Notifier, logrus.Hook
+type shoutrrrTypeNotifier struct {
+	Urls      []string
+	entries   []*log.Entry
+	logLevels []log.Level
+}
+
+func newShoutrrrNotifier(c *cobra.Command, acceptedLogLevels []log.Level) t.Notifier {
+	flags := c.PersistentFlags()
+
+	urls, _ := flags.GetStringArray("notification-url")
+
+	n := &shoutrrrTypeNotifier{
+		Urls:      urls,
+		logLevels: acceptedLogLevels,
+	}
+
+	log.AddHook(n)
+
+	return n
+}
+
+func (e *shoutrrrTypeNotifier) buildMessage(entries []*log.Entry) string {
+	body := ""
+	for _, entry := range entries {
+		body += entry.Time.Format("2006-01-02 15:04:05") + " (" + entry.Level.String() + "): " + entry.Message + "\r\n"
+		// We don't use fields in watchtower, so don't bother sending them.
+	}
+
+	return body
+}
+
+func (e *shoutrrrTypeNotifier) sendEntries(entries []*log.Entry) {
+	// Do the sending in a separate goroutine so we don't block the main process.
+	msg := e.buildMessage(entries)
+	go func() {
+		for _, url := range e.Urls {
+			err := shoutrrr.Send(url, msg)
```

Comment:
If you use the CreateSender approach you can get rid of the for loop and let Shoutrrr handle it for you:
```go
shoutrrr, _ := shoutrrr.CreateSender(e.Urls...);
err := shoutrrr.Send(msg, nil);
```

Reply:
Done. I did not get rid of the loop though. I iterate over the errors now, so the user knows which service url failed.

Reply:
Also I just moved `shoutrrr.CreateSender(e.Urls...)` to the `newShoutrrrNotifier` function so it gets reused.

Reply:
Ah, the Send() returns a slice of errors. Then looping over errors is fine.